### PR TITLE
allows naming - move improve performance file

### PR DESCRIPTION
**Pull request by svittoz**
## Description
In improve_performance function :
- Allows **app_name** parameter so that user can distinguish two kernels from the same project.
- Improve_performance function definition **moved in io folder**.

## Checklist

- [x] If this PR is a bug fix, the bug is documented in the test suite.
- [x] Changes were documented in the changelog (pending section).
- [x] If necessary, changes were made to the documentation (eg new pipeline).

### CODE REVIEW

**Comment by TheooJ** [comment]
Hi Simon !
- "Allows app_name parameter so that user can distinguish two kernels from the same project." -> this is really useful !
- "Lazy operations does not seem to work" which operation are you referring to ? Note that some `ks` operations are rewritten, making them eager: https://github.com/aphp/eds-scikit/blob/053a943fc0266726c9a4b0c547ef5e2ce64e239f/eds_scikit/utils/custom_implem/custom_implem.py#L68 

Also, I think `improve_performance` should be accessible from `eds_scikit`, but shouldn't be written in `__init__.py`. Writing functions there makes it hard to follow the codebase. I suggest writing it elsewhere and making it accessible by setting `from path.to.file import improve_performance` in the `__init__.py`

File: eds_scikit/__init__.py [hunk 1]
```diff
@@ -118,14 +119,18 @@ def improve_performances(
...
-            ("spark.app.name", f"{os.environ.get('USER')}_scikit"),
+            ("spark.app.name", f"{os.environ.get('USER')}_{app_name}_scikit"),
             ("spark.sql.session.timeZone", tz),
             ("spark.sql.execution.arrow.enabled", "true"),
             ("spark.sql.execution.arrow.pyspark.enabled", "true"),
         ]
     )
 
     for key, value in to_add_conf:
+        if not conf.contains(key):
```
Comment (TheooJ) [inline eds_scikit/__init__.py:130]: I'm not sure if we should do this, by default some parameters will not be present in the conf even though they correspond to actual parameters

Comment (svittoz) [reply eds_scikit/__init__.py:130]: You're right. I think it would be useful to check parameters correctness though. I remove it.

**Comment by svittoz** [comment]
> Hi Simon !
> 
> * "Allows app_name parameter so that user can distinguish two kernels from the same project." -> this is really useful !
> * "Lazy operations does not seem to work" which operation are you referring to ? Note that some `ks` operations are rewritten, making them eager: https://github.com/aphp/eds-scikit/blob/053a943fc0266726c9a4b0c547ef5e2ce64e239f/eds_scikit/utils/custom_implem/custom_implem.py#L68
> 
> Also, I think `improve_performance` should be accessible from `eds_scikit`, but shouldn't be written in `__init__.py`. Writing functions there makes it hard to follow the codebase. I suggest writing it elsewhere and making it accessible by setting `from path.to.file import improve_performance` in the `__init__.py`

I moved improve_performance into io folder and import it in the init as you suggested.

File: eds_scikit/__init__.py [hunk 2]
```diff
@@ -39,113 +24,4 @@
...
-        sc.setLogLevel("ERROR")
-
-    sql = spark.sql
-
-    koalas_options()
-
-    return spark, sc, sql
-
-
 koalas_options()
```
Comment (TheooJ) [inline eds_scikit/__init__.py:27]: I would remove this line, as it changes the default behavior of koalas silently which I think could induce bugs. For users that do want to change it, it's done in `improve_performances` so we don't need it here

File: eds_scikit/__init__.py [hunk 3]
```diff
@@ -9,25 +9,10 @@
...
-import pyspark
 from loguru import logger
-from pyspark import SparkContext
-from pyspark.sql import SparkSession
-
-import eds_scikit.biology  # noqa: F401 --> To register functions
-
-import eds_scikit.utils.logging
-
+from .io import koalas_options, improve_performances
```
Comment (TheooJ) [inline eds_scikit/__init__.py:15]: Maybe
```suggestion
from eds_scikit.io import koalas_options, improve_performances
```
to be consistent with `eds_scikit.biology` and `import eds_scikit.utils.logging`, but I think both work

**Review by TheooJ — COMMENTED** [review]
Added some comments, let me know what you think :)

**Review by TheooJ — COMMENTED** [review]
Looks good to me !
Last comment that's a bit nitpick : should we rename the new `improve_performance.py` file ?

**Review by TheooJ — APPROVED** [review]

**Review by svittoz — COMMENTED** [review]
Moving improve_performance out of __init__ and adding app_name parameter.

**Review by svittoz — COMMENTED** [review]
Adding app_name parameter and moving improve_performance out of __init__.py

**Review by svittoz — COMMENTED** [review]
Addin app_name parameter and moving improve_performance.

**Review by Thomzoy — APPROVED** [review]
Ok no problem !
